nixos/modules/services/networking/zeronet.nix

```diff
@@ -0,0 +1,527 @@
+{ config, pkgs, lib, nodes, ... }:
+with lib;
+let
+  cfg = config.services.zeronet;
+
+  ifNotNull = opt: s: if !isNull opt then s else "";
+
+  configFile = builtins.toFile "zeronet.conf" ''
+    [global]
+    data_dir = ${cfg.dataDir}
+    log_dir = ${cfg.logDir}
+  ''
+  + (ifNotNull cfg.uiIP
```

Comment:
Its not needed to implemented all configuration options. Considering many do not have a description I suggest just using `extraConfig` for them.

Reply:
Removed some, feel free to suggest more... I will provide descriptions for the rest then.